Automattic/wp-calypso
# Emails: Add sign up step - initial design

#### Changes proposed in this Pull Request

This PR creates a new sign up step to allow users to add a free trial of Professional Email to their shopping cart when they select a domain during the domains step.

At present, the step is being added to the main `onboarding` flow behind the `signup/professional-email-step` feature flag.

For now, this PR _only_ adds the visuals and minimal wiring for the new step - we will work on wiring the component into more of the flows in subsequent PRs. This PR also doesn't handle the case where the user selects a free WordPress.com subdomain in the domains step.

#### Testing instructions

* Open this branch locally or via calypso.live (the current live link should be in [this comment](https://github.com/Automattic/wp-calypso/pull/54641#issuecomment-881577562))
* Navigate to `/start/onboarding`
* Work through the signup flow, making sure to select a domain registration on the domains step
* Verify that you _don't_ see the email step after selecting a domain registration on the domains step of the flow.
* Navigate to `/start`
* Work through the signup flow, making sure to select a domain registration on the domains step
* Verify that you _don't_ see the email step after selecting a domain registration on the domains step of the flow.
* Navigate to `/start/onboarding?flags=signup/professional-email-step`
* Work through the signup flow and make sure you select a domain registration on the domains step
* Verify that you see the new Professional Email step and that it looks good at various screen sizes and is visually consistent with the previous step in the flow. (Don't test the Add or Skip buttons, as they are currently not wired in.
* Navigate to `/start?flags=signup/professional-email-step`
* Work through the signup flow and make sure you select a domain registration on the domains step
* Verify that you see the new Professional Email step and that it looks good at various screen sizes and is visually consistent with the previous step in the flow. (Don't test the Add or Skip buttons, as they are currently not wired in.

#### Screenshot

<img width="1666" alt="New Professional Email step" src="https://user-images.githubusercontent.com/3376401/126254419-3428a893-6444-4c39-877f-ef106974c410.png">

Original design discussion is in pcDL0W-nj-p2.
